cylc/rose/entry_points.py

```diff
@@ -40,7 +42,23 @@ def pre_configure(srcdir: Path, opts: 'Values') -> dict:
         # nothing to do here
         return {}
 
-    # load the Rose config
+    # If we are validating against source, load saved CLI options
+    # from previous install, as saved in the rose-suite-cylc-install.conf
+    if (
+        getattr(opts, 'against_source', False)
+        and isinstance(opts.against_source, Path)
+    ):
+        opts = retrieve_installed_cli_opts(srcdir, opts)
+    elif (
+        getattr(opts, 'against_source', False)
+        and not isinstance(opts.against_source, Path)
+    ):
+        LOG.warning(
+            'Validate --against-source will not check Rose Options until'
+            ' after installation. (Upgrade to Cylc ≥ 8.3.6 for this feature).'
+        )
```

Comment:
If this branch is not compatible with older versions of cylc-flow, then please up the minimum version requirement rather than logging a warning.

Reply:
It's compatible, but the bug remains unfixed.

Reply:
???